pkg/server/endpoints/node/handler.go

```diff
@@ -480,7 +488,70 @@ func (h *Handler) FetchJWTSVID(ctx context.Context, req *node.FetchJWTSVIDReques
 }
 
 func (h *Handler) PushJWTKeyUpstream(ctx context.Context, req *node.PushJWTKeyUpstreamRequest) (resp *node.PushJWTKeyUpstreamResponse, err error) {
-	return nil, status.Error(codes.Unimplemented, "cannot push JWK upstream")
+	counter := telemetry_server.StartNodeAPIPushJWTKeyUpstreamCall(h.c.Metrics)
+	defer counter.Done(&err)
+	log := h.c.Log.WithField(telemetry.Method, telemetry.PushJWTKeyUpstream)
+
+	_, ok := getPeerCertificate(ctx)
+	if !ok {
+		log.Error("Downstream SVID is required for this request")
+		return nil, status.Error(codes.InvalidArgument, "downstream SVID is required for this request")
+	}
+
+	_, ok = getDownstreamEntry(ctx)
+	if !ok {
+		log.Error("Downstream entry is required for this request")
+		return nil, status.Error(codes.InvalidArgument, "downstream entry is required for this request")
+	}
+
+	err = h.limiter.Limit(ctx, PushJWTKey, 1)
+	if err != nil {
+		log.WithError(err).Error("Rejecting request due to JWK push rate limiting")
+		return nil, status.Error(codes.ResourceExhausted, err.Error())
+	}
+
+	upstreamAuth, useUpstream := h.c.Catalog.GetUpstreamAuthority()
+	if useUpstream { // We are an intermediate SPIRE server
```

Comment:
it is pretty the same than we have on manage.prepareJWTKey() and we are exposiing append to do almost the same, may it worth creating a new method that call append form inside manage, and add all validations the same way we are doing in both places

Reply:
I agree. I think there is an opportunity here to say that the manager is in charge of publishing JWT key's upstream, whether that key originated from a downstream server, or the server itself during rotation. The manager can do the same steps either way:
1) try and publish and warn if unsupported
2) append returned JWT keys into the bundle
3) return the appended bundle (which manager internals can ignore, but this RPC handler can use as the response)

Reply:
The logic in both cases is not exactly the same...and this was done according to the requirements defined on https://github.com/spiffe/spire/issues/1372. But I believe it makes sense to have the same behavior for each case, so I will make the suggested changes anyway. Also this is likely to change due to some issues when dealing with clusters of spire servers that are on a different branch than the one the JWKs traverses.